internal/httpcore/httpcore.go

```diff
@@ -26,6 +26,8 @@ import (
 
 type endpoints struct{}
 
+type middleware struct{}
+
```

Comment:
Since all the middleware implementations are in `middleware.go` I might move this to the top of that file but that's 100% a nit